a little more on Choice of a Hash Algorithm (#207)
oauth-wg/oauth-selective-disclosure-jwt: draft-ietf-oauth-selective-disclosure-jwt.md

## Choice of a Hash Algorithm

For the security of this scheme, the hash algorithm is required to be first and second preimage
resistant, i.e., it is infeasible to calculate the salt and claim value that result in
a particular digest, and it is infeasible to find a different salt and claim value pair that

> result in a matching digest, respectively. Hash algorithms that do not meet the aforementioned
> requirements MUST NOT be used. Note that inclusion in the "Named Information Hash Algorithm"
> registry [@IANA.Hash.Algorithms] alone does not indicate a hash algorithm's suitability for use
> in SD-JWT (it contains several heavily truncated digests, such as
> `sha-256-32` and `sha-256-64`, which are unfit for security applications).

Reviewer: ```suggestion
result in a matching digest, respectively. 

Hash algorithms that do not meet the aforementioned requirements MUST NOT be used. 
Inclusion in the "Named Information Hash Algorithm" registry [@IANA.Hash.Algorithms] 
alone does not indicate a hash algorithm's suitability for use in SD-JWT (it contains several
heavily truncated digests, such as `sha-256-32` and `sha-256-64`, which are unfit for security
applications).
```

## Choice of a Hash Algorithm

> For the security of this scheme, the hash algorithm is required to be first and second preimage

Reviewer 2: Is first and second preimage resistance a common term? I usually see "preimage resistant" and "second-preimage resistant".

Author: ```suggestion
For the security of this scheme, the hash algorithm is required to be preimage resistant and second-preimage 
```

resistant, i.e., it is infeasible to calculate the salt and claim value that result in

> a particular digest, and it is infeasible to find a different salt and claim value pair that
> result in a matching digest, respectively.

Reviewer 2: ```suggestion
a particular digest, and, for any salt and claim value pair, it is infeasible to find a different salt and claim value pair that
result in the same digest, respectively.
```
